app/controllers/messages_controller.rb

```diff
@@ -9,11 +9,15 @@ def new
   def create
     m1 = message_params
     @email = m1[:receiver_id]
-    m1[:receiver_id] = User.find_by_email(@email).id
-    @message = current_user.sent_messages.new(m1)
-    @message.save!
-    if @message.save!
-      redirect_to dashboard_index_path
+    if User.exists?(email: @email)
```

Comment:
We are making two queries here to database.
1. User.exists?
2. User.find

we can simply find user in 1 query and do the operation if it exists